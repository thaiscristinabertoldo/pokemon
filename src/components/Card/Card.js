import React, { memo } from 'react'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPen } from '@fortawesome/free-solid-svg-icons'

import { Button } from '../Button/Button'
import { Container } from '../Container'
import { Image } from '../Image/Image'
import { Tag } from '../Tag/Tag'
import { Title } from '../Title/Title'

import CardStyled from './card.css'

export const Card = memo(({ pokemon: { id, name, image, types = [] }, onEdit }) => {
  return (
    <CardStyled>
      <Image name={name} src={image} />

      <Title title={name} />

      <Container flexDirection="row" margin="0 0 20px 0">
        {types.map((type, index) => {
          return <Tag key={index} text={type} />
        })}
      </Container>

      <Container justifyContent="center" flexDirection="row" margin="10px 0">
        <Button
          text="Editar"
          width="150px"
          prefix={<FontAwesomeIcon icon={faPen} size="1x" swapOpacity />}
          onClick={() => {
            onEdit(id)
          }}
        />
      </Container>
    </CardStyled>
  )
})
