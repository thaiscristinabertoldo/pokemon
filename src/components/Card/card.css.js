import styled from 'styled-components'

import { COLORS } from '../../constants'

const CardStyled = styled.div`
  margin: 0 20px;
  padding: 10px;
  border: none;
  background-color: ${COLORS.white};
  border-radius: 12px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  margin-bottom: 40px;
  &:hover {
    box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
  }
`

export default CardStyled
