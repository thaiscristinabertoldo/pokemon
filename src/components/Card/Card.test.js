import React from 'react'

import { render, cleanup } from '@testing-library/react'

import { Card } from './Card'

afterEach(cleanup)

const pokemon = {
  id: 'UG9rZW1vbjowMDE=',
  image: 'https://img.pokemondb.net/artwork/bulbasaur.jpg',
  name: 'Bulbasaur',
  types: ['Grass', 'Poison'],
}

describe('Card', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Card pokemon={pokemon} />)

    expect(asFragment(<Card />)).toMatchSnapshot()
  })

  it('should take a altText image', () => {
    const { getByAltText } = render(<Card pokemon={pokemon} />)

    expect(getByAltText('Bulbasaur')).toBeTruthy()
  })

  it('should take a tag', () => {
    const { getByText } = render(<Card pokemon={pokemon} />)

    expect(getByText('Grass')).toBeTruthy()
  })
})
