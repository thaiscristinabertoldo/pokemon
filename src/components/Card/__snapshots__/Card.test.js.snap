// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxgMl iPXDbQ"
  >
    <img
      alt="Bulbasaur"
      data-testid="image"
      height="250px"
      src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
      title="Bulbasaur"
      width="250px"
    />
    <h1
      class="sc-AxhUy fribKI"
    >
      Bulbasaur
    </h1>
    <div
      class="sc-AxiKw iKTzNz"
    >
      <div
        class="sc-AxhCb kVJJmU"
      >
        Grass
      </div>
      <div
        class="sc-AxhCb kVJJmU"
      >
        Poison
      </div>
    </div>
    <div
      class="sc-AxiKw eBaALJ"
    >
      <button
        class="sc-AxjAm cGFuEg"
        color="#ffffff"
        data-testid="button"
        width="150px"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-pen fa-w-16 fa-1x fa-swap-opacity "
          data-icon="pen"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
            fill="currentColor"
          />
        </svg>
         Editar
      </button>
    </div>
  </div>
</DocumentFragment>
`;
